src/components/toolbar/tile-toolbar.tsx

```diff
@@ -5,16 +5,47 @@ import { FloatingPortal } from "@floating-ui/react";
 import { Tooltip } from "react-tippy";
 import { useSettingFromStores, useUIStore } from "../../hooks/use-stores";
 import { useTileToolbarPositioning } from "./use-tile-toolbar-positioning";
-import { getToolbarButtonInfo, getDefaultTileToolbarConfig } from "./toolbar-button-manager";
+import { getToolbarButtonInfo, getDefaultTileToolbarConfig, IToolbarButtonInfo } from "./toolbar-button-manager";
 import { useTooltipOptions } from "../../hooks/use-tooltip-options";
 import { TileModelContext } from "../tiles/tile-api";
+import { JSONValue } from "../../models/stores/settings";
 
 interface ToolbarWrapperProps {
   tileType: string,
   readOnly: boolean,
   tileElement: HTMLElement | null,
 }
 
+type IButtonDescription = string | [string, string];
+export function isValidButtonDescription(obj: JSONValue): obj is IButtonDescription {
+  if (!obj) return false;
+  if (typeof obj === 'string') return true;
+  return (obj.constructor === Array
+    && obj.length === 2
+    && typeof obj[0] === 'string'
+    && typeof obj[1] === 'string');
+ }
+
+/**
+ * Create the complete tooltip from the given button information.
+ * Button titles can have placeholders like {1}, {2} which are replaced by button arguments.
+ * If there is a button keyboard shortcut, it is shown after the title.
+ */
+function formatTooltip(desc: IButtonDescription, info: IToolbarButtonInfo) {
```

Comment:
This seems a bit complicated. If I understand it correctly it means the substitution value would have to be included in each of the places where the toolbar button is included. So if we changed how we wanted to call a particular tile in the case of the data set view button, we'd have to update all of the places it was used.

Unless it causes more problems, I'd recommend allowing the components defining the toolbar buttons to set the tooltip. So then it would give the data set view button complete control over this. And it would also allow dynamically changing the tooltip based on the state of the tile. This could be done by moving the `<Tooltip>` inside of `<TileToolbarButton>` instead of out here at the toolbar level.

Reply:
OK, refactored so that tooltips are now created inside of TileToolbarButton.